src/couch_views/test/couch_views_size_test.erl

```diff
@@ -0,0 +1,562 @@
+        {{InitDoc, FinalDoc}, maps:put(DocId, FinalJSONKVs, IdMapAcc)}
+    end, #{}, Transitions),
+
+    {InitDocs, FinalDocs} = lists:unzip(Docs),
+
+    Sig = couch_uuids:random(),
+
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        write_docs(TxDb, Sig, InitDocs)
+    end),
+
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        write_docs(TxDb, Sig, FinalDocs)
+    end),
+
+    validate_index(Db, Sig, IdMap).
+
+
+clear_views(Db) ->
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        #{
+            tx := Tx,
+            db_prefix := DbPrefix
+        } = TxDb,
+        {Start, End} = erlfdb_tuple:range({?DB_VIEWS}, DbPrefix),
+        erlfdb:clear_range(Tx, Start, End),
+
+        GlobalKey = {?DB_STATS, <<"sizes">>, <<"views">>},
+        BinGlobalKey = erlfdb_tuple:pack(GlobalKey, DbPrefix),
+        erlfdb:set(Tx, BinGlobalKey, ?uint2bin(0))
+    end).
+
+
+write_docs(TxDb, Sig, Docs) ->
+    Mrst = #mrst{
+        sig = Sig,
+        views = [#mrview{
+            id_num = 1
+        }]
+    },
+    IdxState = #{
+        last_seq => <<"foo">>
+    },
+    couch_views_indexer:write_docs(TxDb, Mrst, Docs, IdxState).
+
+
+validate_index(Db, Sig, JSONRows) ->
+    #{
+        db_prefix := DbPrefix
+    } = Db,
+    Rows = fabric2_fdb:transactional(Db, fun(TxDb) ->
+        #{
+            tx := Tx
+        } = TxDb,
+        {Start, End} = erlfdb_tuple:range({?DB_VIEWS}, DbPrefix),
+        erlfdb:get_range(Tx, Start, End)
+    end),
+
+    InitAcc = #{
+        row_count => 0,
+        kv_size => 0,
+        ids => #{},
+        rows => []
+    },
+
+    MapData = lists:foldl(fun({Key, Value}, Acc) ->
+        case erlfdb_tuple:unpack(Key, DbPrefix) of
+            {?DB_VIEWS, ?VIEW_INFO, ?VIEW_UPDATE_SEQ, Sig} ->
+                ?assertEqual(<<"foo">>, Value),
+                Acc;
+            {?DB_VIEWS, ?VIEW_INFO, ?VIEW_ROW_COUNT, Sig, 1} ->
+                maps:put(row_count, ?bin2uint(Value), Acc);
+            {?DB_VIEWS, ?VIEW_INFO, ?VIEW_KV_SIZE, Sig, 1} ->
+                maps:put(kv_size, ?bin2uint(Value), Acc);
+            {?DB_VIEWS, ?VIEW_DATA, Sig, ?VIEW_ID_RANGE, DocId, 1} ->
+                [
+                    TotalKeys, TotalSize, UniqueKeys
+                ] = couch_views_encoding:decode(Value),
+                maps:update_with(ids, fun(Ids) ->
+                    false = maps:is_key(DocId, Ids),
+                    maps:put(DocId, {TotalKeys, TotalSize, UniqueKeys}, Ids)
+                end, Acc);
+            {?DB_VIEWS, ?VIEW_DATA, Sig, ?VIEW_MAP_RANGE, 1, MapKey, _DupeId} ->
+                {EncKey, DocId} = MapKey,
+                {UserKey, UserVal} = erlfdb_tuple:unpack(Value),
+
+                UserJsonKey = couch_views_encoding:decode(UserKey),
+                UserJsonVal = couch_views_encoding:decode(UserVal),
+
+                % Assert our encoded sort key is correct
+                EncKey = couch_views_encoding:encode(UserJsonKey, key),
```

Comment:
since we're testing could use an `?assertEqual` here

Reply:
Done.